sklearn/calibration.py

```diff
@@ -236,28 +254,69 @@ def fit(self, X, y, sample_weight=None):
                     warnings.warn("Since %s does not support sample_weights, "
                                   "sample weights will only be used for the "
                                   "calibration itself." % estimator_name)
+            if self.ensemble:
+                # Check that each cross-validation fold can have at least one
+                # example per class
+                if isinstance(self.cv, int):
+                    n_folds = self.cv
+                elif hasattr(self.cv, "n_splits"):
+                    n_folds = self.cv.n_splits
+                else:
+                    n_folds = None
+                if n_folds and np.any([np.sum(y == class_) < n_folds
+                                       for class_ in self.classes_]):
+                    raise ValueError(f"Requesting {n_folds}-fold "
+                                     "cross-validation but provided less than "
+                                     f"{n_folds} examples for at least one "
+                                     "class.")
+                cv = check_cv(self.cv, y, classifier=True)
+
+                for train, test in cv.split(X, y):
+                    this_estimator = clone(base_estimator)
+
+                    if (sample_weight is not None
+                            and base_estimator_supports_sw):
+                        this_estimator.fit(X[train], y[train],
+                                           sample_weight=sample_weight[train])
+                    else:
+                        this_estimator.fit(X[train], y[train])
+
+                    pred_method = get_prediction_method(this_estimator)
+                    preds = _get_predictions(
+                        pred_method, X[test], label_encoder_
+                    )
+
+                    sw = None if sample_weight is None else sample_weight[test]
+                    calibrated_classifier = _fit_calibrator(
+                        this_estimator, preds, y[test], label_encoder_,
+                        self.method, sample_weight=sw
+                    )
+                    self.calibrated_classifiers_.append(calibrated_classifier)
+            else:
+                pred_method = partial(cross_val_predict(
```

Comment:
Let's move `this_estimator = clone(base_estimator)` just above and use `this_estimator` instead of `base_estimator` here